Source/Dafny/Dafny.atg

```diff
@@ -3736,7 +3736,7 @@ UnaryExpression<out Expression e, bool allowLemma, bool allowLambda, bool allowB
 
 /*------------------------------------------------------------------------*/
 PrimaryExpression<out Expression e, bool allowLemma, bool allowLambda, bool allowBitwiseOps>
-= (. e = null; .)
+= (. e = null; Contract.Ensures(Contract.ValueAtReturn(out e) != null); e = dummyExpr; .)
```

Comment:
What is the case in which this returns with a dummyExpr ?

Reply:
When it tries to parse a PrimaryExpression but none of the case pass, e.g. when the token is "multiset" but the next token is not an opening brace "{"

Reply:
To work with the now-defunct CodeContracts toolset, `Contract.Ensures` should be placed immediately inside the enclosing method (together with other `Contract.Requires` and `Contract.Ensures` declaration, but without any preceding code like `e = null;`). Even though we no longer run the CodeContracts tools, I suggest moving this `Contract.Ensures` to be before the `e = null;`.

Reply:
Actually, there's no reason to keep the `e = null;` at all.